SwiftCompilerSources/Sources/Optimizer/Utilities/EscapeInfoVisitor.swift

```diff
@@ -0,0 +1,633 @@
+          return walkDown(address: rea, path: newPath, state: state.with(knownType: nil))
+        }
+      case let pb as ProjectBoxInst:
+        if let newPath = pop(.classField, index: pb.fieldIndex, from: path, yielding: pb) {
+          return walkDown(address: pb, path: newPath, state: state.with(knownType: nil))
+        }
+      case let se as SwitchEnumInst:
+        if let (caseIdx, path) = path.pop(kind: .enumCase) {
+          if let succBlock = se.getUniqueSuccessor(forCaseIndex: caseIdx) {
+            if let payload = succBlock.arguments.first {
+              return walkDown(value: payload, path: path, state: state)
+            }
+          }
+        } else if path.topMatchesAnyValueField {
+          // We don't know the enum case: we have to containue with _all_ cases.
+          for succBlock in se.block.successors {
+            if let payload = succBlock.arguments.first {
+              let walkResult = walkDown(value: payload, path: path, state: state)
+              if walkResult.next == .abortWalk {
+                return walkResult
+              }
+            }
+          }
+        } else {
+          return (state, .abortWalk)
+        }
+        // MARK: destruction instructions
+      case is StoreInst, is StoreWeakInst, is StoreUnownedInst:
+        let store = instruction as! StoringInstruction
+        if operand == store.sourceOperand {
+          return walkUp(address: store.destination, path: path, state: state)
+        } else {
+          assert(operand == store.destinationOperand)
+          if let si = store as? StoreInst, si.destinationOwnership == .assign {
+            if handleDestroy(of: operand.value, path: path, followStores: state.followStores, knownType: nil) {
+              return (state, .abortWalk)
+            }
+          }
+          if state.followStores {
+            return walkUp(value: store.source, path: path, state: state)
+          }
+        }
+      case let copyAddr as CopyAddrInst:
+        if canIgnoreForLoadOrArgument(path) { return (state, .stopWalk) }
+        if operand == copyAddr.sourceOperand {
+          return walkUp(address: copyAddr.destination, path: path, state: state)
+        } else {
+          if !copyAddr.isInitializationOfDest {
+            if handleDestroy(of: operand.value, path: path, followStores: state.followStores, knownType: nil) {
+              return (state, .abortWalk)
+            }
+          }
+          
+          if state.followStores {
+            assert(operand == copyAddr.destinationOperand)
+            return walkUp(value: copyAddr.source, path: path, state: state)
+          }
+        }
+      case is DestroyValueInst, is ReleaseValueInst, is StrongReleaseInst, is DestroyAddrInst:
+        if handleDestroy(of: operand.value, path: path, followStores: state.followStores, knownType: state.knownType) {
+          return (state, .abortWalk)
+        }
+      case is ReturnInst:
+        return (state, .abortWalk)
```

Comment:
Can you wrap the return value in a function call (`isEscaping`) here and everywhere the walk is aborted due to an escape point? Like it's done in the current EscapeInfo implementation.
This is super useful for debugging, because you can set a single breakpoint on `isEscaping`.